### PR TITLE
Move from `mocha` to `vitest`

### LINKED ISSUE #585: Move to `vite`
We are using `@bentley/react-scripts` which is based on create react app, however this way of building apps have been dropped by React and is no longer recommended.

`vite` appear to be the goto solution and is already used in core and studio, so we should probably create a test app using vite in the first time to validate that everything works correctly, and in a second phase probably retire the current cra based test apps.

Using `vitest` might also help us with up to date testing solution rather than mocha which is getting old and makes testing with iTwinUI difficult.

```[tasklist]
### Tasks
- [x] Create `vite` based test apps
- [ ] Explore `vitest` as a replacement for `mocha`
- [x] Remove `cra` based test apps.
```

### PR DESCRIPTION
**Pull request by GerardasB**
## Changes

This PR fixes #585 by replacing `mocha` with `vitest`.
No additional assertion libraries are used: `sinon`, `chai` are removed.
Since `v8` is used for coverage, current `istanbul ignore` lines are ignored causing coverage to drop. These will have to be removed in a separate PR.
Some of the tests were skipped, these are marked with `TODO: vitest`.
`cobertura` code coverage is no longer published, only `junit` report.

## Testing

N/A

### CODE REVIEW

**Review by saskliutas — APPROVED** [review]

File: ui/appui-react/src/test/pickers/ListPicker.test.tsx [hunk 1]
```diff
@@ -110,19 +96,20 @@ describe("ListPicker", () => {
         />
       </Provider>
     );
+    x.debug();
```
[the remark this answers is not in the document]
Comment (GerardasB) [reply ui/appui-react/src/test/pickers/ListPicker.test.tsx]: Removed.

File: ui/components-react/src/test/filter-builder/Operators.test.ts [hunk 2]
```diff
@@ -100,10 +95,10 @@ describe("getPropertyFilterBuilderOperators", () => {
...
   it("returns correct values", () => {
-    expect(isUnaryPropertyFilterOperator("is-true")).to.be.true;
-    expect(isUnaryPropertyFilterOperator("is-false")).to.be.true;
-    expect(isUnaryPropertyFilterOperator("is-null")).to.be.true;
-    expect(isUnaryPropertyFilterOperator("is-not-null")).to.be.true;
+    expect(isUnaryPropertyFilterOperator("is-true")).toEqual(true);
+    expect(isUnaryPropertyFilterOperator("is-false")).toEqual(true);
+    expect(isUnaryPropertyFilterOperator("is-null")).toEqual(true);
+    expect(isUnaryPropertyFilterOperator("is-not-null")).toEqual(true);
     expect(isUnaryPropertyFilterOperator("is-equal")).to.be.false;
```
[the remark this answers is not in the document]
Comment (GerardasB) [reply ui/components-react/src/test/filter-builder/Operators.test.ts]: I've replaced-all those assertions that were causing build issues. There are still a number of chai assertions left..

[the remark this answers is not in the document]
Comment (GerardasB) [reply ui/components-react/src/test/filter-builder/Operators.test.ts]: Replaced `false` assertions w/ vitest assertions.

File: .github/workflows/CI.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20.x
+
+      - name: Rush install
+        run: node common/scripts/install-run-rush.js install --variant ${{ matrix.variant }}
+
+      - name: Rush build
```
Comment (saskliutas) [inline .github/workflows/CI.yaml:37]: Shouldn't `lint` and `prettier` steps also be included in this workflow?

**Review by saskliutas — APPROVED** [review]

File: .github/workflows/CI.yaml (hunk 3 above)
Comment (GerardasB) [reply .github/workflows/CI.yaml:37]: Eventually - yes. I tried to keep this minimal and related to unit-tests only.

**Pull request merged**